src/selectors/metadata.js

```diff
@@ -175,6 +175,27 @@ export const customProfileTypesSelector = memoize(() =>
 	),
 );
 
+export const mappedDefinitionsSelector = memoize(moduleName =>
+    createSelector(
+        definitionsModule(moduleName),
+        currentLocaleOrDefault,
+        (immutableProductDefinition, locale, key, defaultValue = key) => {
+            const productDefinitions = immutableProductDefinition.toJS();
+            const propertyMap = [];
+            each(productDefinitions, productDefinition => {
+                const newObj = {
+                    name: getLocalization(productDefinition?.displayName, locale, defaultValue),
+                    type: productDefinition.isSharedEntity === true ? "Shared" : "Embedded",
+                    isBuiltIn: productDefinition.isBuiltIn,
+                    description: productDefinition.description ?? "",
+                };
+                propertyMap.push(newObj);
+            });
+            return propertyMap;
```

Comment:
I think it can be simplified like this:

```
export const mappedDefinitionsSelector = memoize(moduleName =>
	createSelector(
		definitionsModule(moduleName),
		currentLocaleOrDefault,
		(definitions, locale, key, defaultValue = key) => {
			return definitions.map(definition => {
				return definition
					.set("type", definition.isSharedEntity === true ? definitionType .Shared : definitionType.Embedded)
					.set("name", getLocalization(definition?.displayName, locale, defaultValue));
			})
		},
	),
);
```

In file ~\src\constants.js define new const definitionType
```
export const definitionType = {
	Shared: "Shared",
	Embedded: "Embedded"
};
```